### DISCUSSION TITLE
Extending the tour service leads to unregistered anchor ids

### DISCUSSION THREAD

**Discussion by Shaent** [hakimio/ngx-ui-tour#48] [Q&A] [answered]
Hi,

in my Angular Application I wrote a new Service which extends the TourService so that I can add some additional methods I need.

Extending the service does not work as expected tough. Using the inherited methods like `this.tourService.initialize()` or `this.tourService.startAt()` works but in the end the service always fails with: _Can't attach to unregistered anchor with id app-home_ in the `showStep()` method because `this.anchors` seems to be empty.

During debugging I realized that somehow the information about registered anchor IDs is missing.

I set a a minimal example using your lazy-loaded-tour example which you can find in [this repo](https://github.com/Shaent/ng-tour-test). It is also setup for debugging with VSC if this helps.

Is this a bug or am I doing something wrong regarding the inheritance?

**Comment by hakimio** [comment, marked as answer]
The `TourService` class is not supposed to be extended. You have to use dependency injection like demonstrated [here](https://hakimio.github.io/ngx-ui-tour/md-menu/FAQ#disable-scrolling).
